libcloud/dns/drivers/godaddy.py

```diff
@@ -0,0 +1,521 @@
+# Licensed to the Apache Software Foundation (ASF) under one or more
+# contributor license agreements.  See the NOTICE file distributed with
+# this work for additional information regarding copyright ownership.
+# The ASF licenses this file to You under the Apache License, Version 2.0
+# (the "License"); you may not use this file except in compliance with
+# the License.You may obtain a copy of the License at
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+__all__ = [
+    'GoDaddyDNSDriver'
+]
+
+try:
+    import simplejson as json
+except:
+    import json
+
+from libcloud.common.base import ConnectionKey, JsonResponse
+from libcloud.common.types import LibcloudError
+from libcloud.utils.py3 import httplib
+from libcloud.dns.types import Provider, RecordType
+from libcloud.dns.base import DNSDriver, Zone, Record
+
+API_ROOT = 'https://api.godaddy.com/'
+VALID_RECORD_EXTRA_PARAMS = ['prio', 'ttl']
+
+
+class GoDaddyDNSException(LibcloudError):
+    def __init__(self, code, message):
+        self.code = code
+        self.message = message
+        self.args = (code, message)
+
+    def __str__(self):
+        return self.__repr__()
+
+    def __repr__(self):
+        return '<GoDaddyDNSException in %d: %s>' % (self.code, self.message)
+
+
+class GoDaddyDNSResponse(JsonResponse):
+    valid_response_codes = [httplib.OK, httplib.ACCEPTED, httplib.CREATED,
+                            httplib.NO_CONTENT]
+
+    def parse_body(self):
+        if not self.body:
+            return None
+        # json.loads doesn't like the regex expressions used in godaddy schema
+        self.body = self.body.replace('\\.', '\\\\.')
```

Comment:
https://www.google.com.au/webhp?sourceid=chrome-instant&ion=1&espv=2&ie=UTF-8#q=json%20loads%20invalid%20escape yeah I'm not the only one. the other option is to introduce cjson, but adding another prereq to the project for just one regex statement seems excessive